apps/condo/domains/news/schema/NewsItemUserRead.test.js

```diff
@@ -0,0 +1,291 @@
+                })
+            })
+        })
+
+        describe('update', () => {
+            test('admin can', async () => {
+                const [newsItem] = await createTestNewsItem(adminClient, dummyO10n)
+                const [objCreated] = await createTestNewsItemUserRead(adminClient, newsItem, adminClient.user)
+                const [obj, attrs] = await updateTestNewsItemUserRead(adminClient, objCreated.id)
+
+                expect(obj.dv).toEqual(1)
+                expect(obj.sender).toEqual(attrs.sender)
+                expect(obj.v).toEqual(2)
+                expect(obj.updatedBy).toEqual(expect.objectContaining({ id: adminClient.user.id }))
+            })
+
+            test('support can', async () => {
+                const [newsItem] = await createTestNewsItem(adminClient, dummyO10n)
+                const [objCreated] = await createTestNewsItemUserRead(adminClient, newsItem, supportClient.user)
+                const [obj, attrs] = await updateTestNewsItemUserRead(supportClient, objCreated.id)
+
+                expect(obj.id).toMatch(UUID_RE)
+                expect(obj.dv).toEqual(1)
+                expect(obj.sender).toEqual(attrs.sender)
+                expect(obj.v).toEqual(2)
+                expect(obj.updatedBy).toEqual(expect.objectContaining({ id: supportClient.user.id }))
+            })
+
+            test('stuff can\'t', async () => {
+                const [newsItem] = await createTestNewsItem(adminClient, dummyO10n)
+                const [objCreated] = await createTestNewsItemUserRead(adminClient, newsItem, adminClient.user)
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    const [obj, attrs] = await updateTestNewsItemUserRead(staffClient, objCreated.id)
+                })
+            })
+
+            test('resident can\'t', async () => {
+                const [newsItem] = await createTestNewsItem(adminClient, dummyO10n)
+                const [objCreated] = await createTestNewsItemUserRead(adminClient, newsItem, residentClient.user)
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestNewsItemUserRead(residentClient, objCreated.id)
+                })
+            })
+
+            test('anonymous can\'t', async () => {
+                const [newsItem] = await createTestNewsItem(adminClient, dummyO10n)
+                const [objCreated] = await createTestNewsItemUserRead(adminClient, newsItem, adminClient.user)
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await updateTestNewsItemUserRead(anonymousClient, objCreated.id)
+                })
+            })
+        })
+
+        describe('hard delete', () => {
+            test('admin can\'t', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await NewsItemUserRead.delete(adminClient, dummyNewsItem.id)
+                })
+            })
+
```

Comment:
what about supports? :)

Reply:
I do not think it's needed. No one can hard delete.